frontend/src/metabase/actions/components/ActionForm/FormFieldEditor.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useMemo } from "react";
+import { t } from "ttag";
+
+import Radio from "metabase/core/components/Radio";
+
+import type { ActionFormFieldProps } from "metabase/actions/types";
+import type { FieldSettings, FieldType } from "metabase-types/api";
+
+import { FieldSettingsButtons } from "../../containers/ActionCreator/FormCreator/FieldSettingsButtons";
+import {
+  getFieldTypes,
+  getInputTypes,
+} from "../../containers/ActionCreator/FormCreator/constants";
+
+import { FormFieldWidget } from "./ActionFormFieldWidget";
+import {
+  Column,
+  DragHandle,
+  EditorContainer,
+  FormFieldContainer,
+  Header,
+  InputContainer,
+  Title,
+  Subtitle,
+  PreviewContainer,
+} from "./FormFieldEditor.styled";
+
+export interface FormFieldEditorProps {
+  field: ActionFormFieldProps;
+  fieldSettings: FieldSettings;
+  isEditable: boolean;
+  onChange: (settings: FieldSettings) => void;
+}
+
+function FormFieldEditor({
+  field,
+  fieldSettings,
+  isEditable,
+  onChange,
+}: FormFieldEditorProps) {
+  const fieldTypes = useMemo(getFieldTypes, []);
+  const inputTypes = useMemo(getInputTypes, []);
+
+  const handleChangeFieldType = (nextFieldType: FieldType) => {
+    const [defaultInputType] = inputTypes[nextFieldType];
+    onChange({
+      ...fieldSettings,
+      fieldType: nextFieldType,
+      inputType: defaultInputType.value,
+    });
+  };
+
+  return (
+    <FormFieldContainer>
+      <EditorContainer>
+        <Column>{isEditable && <DragHandle name="grabber2" />}</Column>
+        <Column full>
+          <Header>
+            <Title>{field.title}</Title>
+            {isEditable && (
+              <FieldSettingsButtons
+                fieldSettings={fieldSettings}
+                onChange={onChange}
+              />
+            )}
+          </Header>
+          {isEditable && fieldSettings && (
+            <>
+              <Subtitle>{t`Field type`}</Subtitle>
+              <Radio
+                value={fieldSettings.fieldType}
+                options={fieldTypes}
+                aria-label={t`Field type`}
+                variant="bubble"
+                onChange={handleChangeFieldType}
+              />
+            </>
+          )}
```

Comment:
There's a nasty thing that `fieldSettings` are `undefined` for a moment when you add a new parameter. This happens because field settings are created in a `useEffect`. I believe we should patch the `QueryActionContextProvider` to create default field settings immediately when a new parameters gets added, but this feels as a good TODO for a bit broader refactoring PR